library/ratelimiting/getRateLimitedEndpoint.ts

```diff
@@ -0,0 +1,26 @@
+import { Endpoint } from "../agent/Config";
+import { Context } from "../agent/Context";
+import { ServiceConfig } from "../agent/ServiceConfig";
+
+export function getRateLimitedEndpoint(
+  context: Readonly<Context>,
+  config: ServiceConfig
+): Endpoint | undefined {
+  const matches = config
+    .getEndpoints(context)
+    .filter((m) => m.rateLimiting && m.rateLimiting.enabled);
+
+  if (matches.length === 0) {
+    return undefined;
+  }
+
+  matches.sort((a, b) => {
+    if (a.rateLimiting.windowSizeInMS === b.rateLimiting.windowSizeInMS) {
```

Comment:
If we have two matching routes, e.g.:
- /test/* - max. 5 requests in 30s
- /test/v2/* - max. 2 requests in 40s

In this case we will prefer the request with the smaller windows size? Maybe we should use a metric that includes the window size and max requests?